### DISCUSSION TITLE
Renovate links to wrong users in release notes on platforms other than GitHub

### DISCUSSION THREAD

**Discussion by elchenberg** [renovatebot/renovate#30219] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Platform: GitLab (private/self-hosted) Renovate: 37.432.0

### Please tell us more about your question or problem

When users of our self-hosted GitLab instance are mentioned in the release notes of our projects Renovate turns them into GitHub links that might or might not exist:

Line from the original release notes:

```md
- "commit message" by @user.name`
```

Line that was added by Renovate to the release notes of a merge request:

```md
- "commit message" by [@&#8203;user](https://github.com/user).name
```

I guess this happens because the linkify function uses remark-github:

```js
import remark from 'remark';
import github from 'remark-github';
// ...
export async function linkify(
  content: string,
  options: github.RemarkGithubOptions,
): Promise<string> {
  // https://github.com/syntax-tree/mdast-util-to-markdown#optionsbullet
  const output = await remark()
    .use({ settings: { bullet: '-' } })
    .use(github, { mentionStrong: false, ...options })
    .process(content);
  return output.toString();
}
```

See https://github.com/renovatebot/renovate/blob/37.432.0/lib/util/markdown.ts#L34

I could not find out how to change this behavior so I think I might have found a bug.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment]
I think we only want ot support linking incase of `github` repo. See here:

https://github.com/renovatebot/renovate/blob/4b502025fedfbfa62379147445ccebff0144e3d0/lib/workers/repository/update/pr/changelog/release-notes.ts#L201-L206

Theself-hosted gitlab instance must have a different `baseUrl`. We will need to use another diff mathcing to exclude gitlab repos which covers self-hosted instances too. Maybe invert it to only linkify github repos instead.

**Reply by HonkingGoose** [reply]
This issue seems related:

- https://github.com/renovatebot/renovate/issues/19503

**Reply by HonkingGoose** [reply]
@rarkins what do you want me to do?

- Comment in #19503 about how Renovate always uses the GitHub _username_?
- Or, create new issue just for the GitHub username problem.

**Reply by viceice** [reply]
I think we should make the replacement more flexible. we already know the target platform, so we could make all links absolute.

after that we can easily replace GitHub links with our helper url.
